partner_financial_risk/models/res_partner.py

```diff
@@ -0,0 +1,150 @@
+        max_date = self._max_risk_date_due()
+        for partner in self:
+            invoices_out = partner.invoice_ids.filtered(
+                lambda x: x.type == 'out_invoice')
+            invoices = invoices_out.filtered(
+                lambda x: x.state in ['draft', 'proforma', 'proforma2'])
+            partner.risk_invoice_draft = sum(invoices.mapped('amount_total'))
+            invoices = invoices_out.filtered(
+                lambda x: x.state == 'open' and x.date_due >= max_date)
+            partner.risk_invoice_open = sum(invoices.mapped('residual'))
+            invoices = invoices_out.filtered(
+                lambda x: x.state == 'open' and x.date_due < max_date)
+            partner.risk_invoice_unpaid = sum(invoices.mapped('residual'))
+
+    @api.multi
+    @api.depends('credit', 'risk_invoice_open', 'risk_invoice_unpaid')
+    def _compute_risk_account_amount(self):
+        for partner in self:
+            partner.risk_account_amount = (
+                partner.credit - partner.risk_invoice_open -
+                partner.risk_invoice_unpaid)
+
+    @api.multi
+    @api.depends(lambda x: x._get_depends_compute_risk_exception())
+    def _compute_risk_exception(self):
+        risk_field_list = self._risk_field_list()
+        for partner in self:
+            amount = 0.0
+            for risk_field in risk_field_list:
+                field_value = getattr(partner, risk_field[0], 0.0)
+                max_value = getattr(partner, risk_field[1], 0.0)
+                if max_value and field_value > max_value:
+                    partner.risk_exception = True
+                if getattr(partner, risk_field[2], False):
+                    amount += field_value
+            partner.risk_total = amount
+            if amount > partner.credit_limit:
+                partner.risk_exception = True
+
+    @api.model
+    def _max_risk_date_due(self):
+        return fields.Date.to_string(datetime.today().date() - relativedelta(
+            days=self.env.user.company_id.invoice_due_margin))
+
+    @api.model
+    def _risk_field_list(self):
+        return [
+            ('risk_invoice_draft', 'risk_invoice_draft_limit',
+             'risk_invoice_draft_include'),
+            ('risk_invoice_open', 'risk_invoice_open_limit',
+             'risk_invoice_open_include'),
+            ('risk_invoice_unpaid', 'risk_invoice_unpaid_limit',
+             'risk_invoice_unpaid_include'),
+            ('risk_account_amount', 'risk_account_amount_limit',
+             'risk_account_amount_include'),
+        ]
+
+    @api.model
+    def _get_depends_compute_risk_exception(self):
+        # TODO: Improve code without lose performance
+        tuple_list = self._risk_field_list()
+        res = [x[0] for x in tuple_list]
+        res.extend([x[1] for x in tuple_list])
+        res.extend([x[2] for x in tuple_list])
+        res.append('credit_limit')
+        return res
+
+    @api.model
+    def process_due_invoice(self):
```

Comment:
Call this method `process_unpaid_invoices` for consistency (in plural, as you're processing all the unpaid, not only one). Singular form is only used in model names and module names, not in methods.